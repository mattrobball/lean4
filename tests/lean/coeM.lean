/-
# Testing monad lift coercion elaborator

The functions inserted for the coercions are supposed to be inlined immediately during elaboration.
-/

variable (p : Nat â†’ Prop) (m : IO (Subtype p))

/-!
`Lean.Internal.liftCoeM`
-/
#check (m : (ReaderT Int IO) Nat)

/-!
`Lean.Internal.coeM`
-/
#check (m : IO Nat)
