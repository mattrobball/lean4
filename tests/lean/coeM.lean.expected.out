do
  let a ← liftM m
  pure a.val : ReaderT Int IO Nat
do
  let a ← m
  pure a.val : EIO IO.Error Nat
